GVFS/GVFS/CommandLine/CloneVerb.cs

```diff
@@ -567,6 +567,27 @@ private void CheckNotInsideExistingRepo(string normalizedEnlistmentRootPath)
             }
 
             GitProcess.Result forceCheckoutResult = git.ForceCheckout(branch);
+            if (forceCheckoutResult.HasErrors)
+            {
+                // It is possible to have the above TryDownloadCommit() fail because we
+                // already have the commit and root tree we intend to check out, but
+                // don't have a tree further down the working directory. If we fail
+                // checkout here, it may be due to not having these trees and the
+                // read-object hook is not available yet. Force downloading the commit
+                // again and retry the checkout.
+
+                if (!this.ForceTryDownloadCommit(

```

Comment:
I don't think we should just re-download on all errors. We know exactly when this specific error happens because the error message says something about missing trees. I would check for that and only redownload/re-checkout in that case.

Reply:
@sanoursa @jrbriggs How much are we dependent on the installed language (for Git)? If we search for a specific string, then we will be broken if someone installs using a different locale. Or, is there a way to force the language in a specific call?

Reply:
That's a great question, but we're already dependent on some error strings. E.g. see the very next section in this method, we're checking the checkout errors for certain errors that we ignore and consider non-errors.

Reply:
I can add an extra check here, if we are already tied to the message. We will need to consider multi-language support at some time, and working with a non-English Git will be important before we localize VFS for Git.